Projectoid.js

```diff
@@ -0,0 +1,313 @@
+    }
+  }
+
+  let requestData = {
+    path: "telegram/botpanel/broadcast.php",
+    method: "post",
+    body: {
+      access_token,
+      bot_token: bot.token,
+      admin: user.telegramid,
+      method: "forwardMessage",
+      from_chat_id,
+      message_id,
+      protectContent,
+      webhookUrl,
+    },
+    onSuccess: cmd,
+  };
+  apiCall(requestData);
+}
+
+function initiateCopyBroadcast(options) {
+  if (!options) {
+    throw libPrefix + ": initiateCopyBroadcast: options not found";
+  }
+  let from_chat_id = options.from_chat_id;
+  let message_id = options.message_id;
+
+  if (!from_chat_id || !message_id) {
+    throw libPrefix + ": initiateCopyBroadcast: chat id or message id was not found";
+  }
+  let protectContent = !options.protectContent ? false : options.protectContent;
+  let webhookUrl = !options.webhookUrl ? null : options.webhookUrl;
+  let cmd = !options.command ? null : options.command;
+
+  if (!options.access_token) {
+    var access_token = Bot.getProperty("Projectoid_AccessToken");
+    if (!access_token) {
+      throw libPrefix + ": initiateCopyBroadcast: Bot Access Token Not Found";
+    }
+  }
+
+  let requestData = {
+    path: "telegram/botpanel/broadcast.php",
+    method: "post",
+    body: {
+      access_token,
+      bot_token: bot.token,
+      method: "copyMessage",
+      admin: user.telegramid,
+      from_chat_id,
+      message_id,
+      protectContent,
+      webhookUrl,
+    },
+    onSuccess: cmd,
+  };
+  apiCall(requestData);
+}
+//---------------------Broadcast Functions Ends------------------------------
+
+/**
+ * checkMembership() - Checks membership of users in chats.
+ *
+ * @since 1.0.0
+ *
+ * @param {object} options - The membership check options.
+ */
+function checkMembership(options) {
+  if (!options) {
+    throw libPrefix + ": checkMembership: params not found";
+  }
+  if (!options.userId) {
+    throw libPrefix + ": checkMembership: User id Not Found";
+  }
+  if (!options.chats) {
+    throw libPrefix + ": checkMembership: Chat IDs Not Found";
+  }
+  if (!options.command) {
+    throw libPrefix + ": checkMembership: command not found to return response";
+  }
+
+  let requestData = {
+    path: "telegram/membership/index.php",
+    method: "post",
+    body: {
+      token: bot.token,
+      userId: options.userId,
+      chatIds: options.chats,
+    },
+    onSuccess: options.command,
+  };
+  apiCall(requestData);
+}
+
+// Function called when an API answer is received
+function onApiAnswer() {
+  // Parse the content of the response, which is in JSON format
+  let options = content;
+  try {
+    options = JSON.parse(options);
+  } catch (error) {}
+
+  if (!params || params === null || params === "null") {
+  } else {
+    Bot.runCommand(params, options);
+  }
+}
+
+// Function called when an API request results in an error
+function onApiError() {
+  throw libPrefix + ": " + content + "\nGet Help at @ProjectoidChat";
+}
+
+publish({
+  apiCall: apiCall,
```

Comment:
descriptions is needed below all methods here. 

hard to understand why we need this methods